### DISCUSSION TITLE
Through scripting, would it be possible to reproduce the accordion layout from XMonad?

### DISCUSSION THREAD

**Discussion by Merulox** [i3/i3#5772] [Q&A] [answered]
I have no experience with scripting but I'd be willing to learn to make this a reality-- I'd just like to know beforehand whether or not it would be possible.

**Comment by okraits** [comment]
Could you please explain what the accordion layout is? Somebody not familiar with XMonad doesn't know that.

**Reply by Merulox** [reply]
My bad.
![image](https://github.com/i3/i3/assets/68356319/b9059b30-4af3-4ed8-9b3f-3c4e7968e6a4)  
and
![image](https://github.com/i3/i3/assets/68356319/2a6aaa3a-983a-4d88-b0fa-3a6a70d01ff5)

It looks a little like this: whichever window you focus on will expand so as to be usable. It's kind of similar to i3's stacked layout except you can see all the windows at all times.

**Reply by okraits** [reply, marked as answer]
Yes, I think it's possible. You need a script which listens for window focus events (https://i3wm.org/docs/ipc.html#_window_event) and then resizes the windows accordingly - make the focused window taller and shrink the other windows on the focused workspace.

**Reply by orestisfl** [reply]
Probably all window events, to react to new windows and closed windows as well.
